### ISSUE TITLE
return more than just error

### ISSUE THREAD

**Issue by u007** [graph-gophers/graphql-go#259]
i have a mutation that checks for invalid fields or missing data in the fields,

when i return error, the resolver seems to only return error message and path,
how do i return more values like which fields and what are the errors on respective fields?

```
func (r *helloWorldResolver) Hello(ctx context.Context) (string, error) {
	return "Hello world!", fmt.Errorf("single error here")
}
```

**Comment by pavelnikolov** [comment]
I have never had to worry about this. The GraphQL spec seems to allow this 🤷‍♂️  
I do not plan to implement this any time soon, however, PRs are welcome.

**Comment by technoweenie** [comment]
Have you considered returning a custom error type in your mutation's response type? Validation messages are not what I would consider an application _error_. Things are functioning as intended if the app rejected a new User object if the client doesn't send the username, for example. These model or form validation errors are likely something you want displayed in your client.

Here's a sample gql schema to illustrate. This isn't something I've personally done with GraphQL (yet), but is a pattern I've used in REST APIs.

```graphql
type Mutation {
  createWidget(input: CreateWidgetInput!): CreateWidgetPayload
}

enum ErrorType {
  MISSING
  BAD_FORMAT
}

type Error {
  field: String!
  kind: ErrorType!
  messages: [String!]!
}

type CreateWidgetPayload {
  widget: Widget
  errors: [Error!]!
}
```

I found a few GraphQL blog posts supporting this idea: [GraphQL Mutations and Form Errors](http://blog.rstankov.com/graphql-mutations-and-form-errors/) shows a direct form validation example, and [The Definitive Guide to Handling GraphQL Errors](https://itnext.io/the-definitive-guide-to-handling-graphql-errors-e0c58b52b5e1) goes more into _why_.
